### PR TITLE
add intermediate result tests to arbitrary configurations

### PR DESCRIPTION
**Pull request by onderkalaci**
add intermediate result tests to arbitrary configurations

### CODE REVIEW

File: src/test/regress/expected/intermediate_result_pruning_queries_1.out [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+-- and the final query is another router query on another worker
+WITH some_values_1 AS MATERIALIZED
+	(SELECT key, random() FROM table_1 WHERE value IN ('3', '4')),
+	some_values_2 AS MATERIALIZED
+	(SELECT key, random() FROM some_values_1 JOIN table_2 USING (key) WHERE table_2.key = 1)
+SELECT
+	count(*)
+FROM
+	(some_values_2 JOIN table_2 USING (key)) JOIN some_values_1 USING (key) WHERE table_2.key = 3;
+ count
```
Comment (onderkalaci) [inline src/test/regress/expected/intermediate_result_pruning_queries_1.out:100]: can we have non-zero results?

Comment (onderkalaci) [reply src/test/regress/expected/intermediate_result_pruning_queries_1.out:100]: done

**Comment by SaitTalhaNisanci** [comment]
I wanted to see the failure reason on 14-check-arbitrary-configs but after storing the logfiles, I am not able to get the same failure, it feels like a heisenbug :)

**Review by SaitTalhaNisanci — APPROVED** [review]
I couldn't get the flaky failure even though I reran multiple times, we can investigate that in the main PR if that happens

File: src/test/regress/custom_sql_schedule [hunk 2]
```diff
@@ -3,3 +3,4 @@ test: prepared_statements_1 prepared_statements_2 prepared_statements_3
 test: prepared_statements_4
 test: ch_benchmarks_1 ch_benchmarks_2 ch_benchmarks_3
 test: ch_benchmarks_4 ch_benchmarks_5 ch_benchmarks_6
+test: intermediate_result_pruning_queries_1 intermediate_result_pruning_queries_2
```
Comment (SaitTalhaNisanci) [inline src/test/regress/custom_sql_schedule:6]: Minor: I skipped style related things as we will fix them in the main PR (such as no new line)

**Pull request merged**